pkg/controller/migmigration/task.go

```diff
@@ -1213,6 +1366,32 @@ func (t *Task) hasVerify() bool {
 	return t.Owner.Spec.Verify
 }
 
+// Returns true if the IndirectImageMigration override on the plan is set (plan is configured not to do direct migration)
+func (t *Task) indirectImageMigration() bool {
+	return t.PlanResources.MigPlan.Spec.IndirectImageMigration
+}
+
+// Returns true if the IndirectImageMigration override on the plan is not set (plan is configured to do direct migration)
+func (t *Task) directImageMigration() bool {
+	return !t.indirectImageMigration()
+}
+
+// Returns true if the IndirectVolumeMigration override on the plan is set (plan is configured not to do direct migration)
+func (t *Task) indirectVolumeMigration() bool {
+	return t.PlanResources.MigPlan.Spec.IndirectVolumeMigration
+}
+
+// Returns true if the IndirectVolumeMigration override on the plan is not set (plan is configured to do direct migration)
+// There must exist a set of direct volumes for this to return true
+func (t *Task) directVolumeMigration() bool {
+	return !t.indirectVolumeMigration() && t.hasDirectVolumes()
```

Comment:
So one concern I have adding this here is it means we end up generating the direct volume migration list more than once. `hasDirectVolumes` iterates over the whole PV list, figures out which ones are needed, and then figures out access modes, etc. and builds the list, where we use it here to determine whether the list is empty or not. Then we throw it away and buildDirectVolumeMigration calls it again. We then call it again when deciding whether we need to wait on the DVM. I think it would be more efficient to just go to the createDVM step if the boolean is false, and then there we call hasDirectVolumes() and if it's empty, no DVM is created. Then, when we get to the wait step, we don't have to iterate over the PVC list yet again, we can just do the get for the DVM, and if it's not found, we're done -- there's nothing to wait for, so we call `t.next()`.

Reply:
Great points here. This needs to be addressed.

Reply:
Opening follow-up issue.

Reply:
https://github.com/konveyor/mig-controller/issues/795